### PR TITLE
Fix ProjectCollection.search; use in find_collection

### PR DESCRIPTION
**Pull request by maxhutch**
# Citrine Python PR

## Description 

This adds a `get_resource_post` method to the session that behaves
like get_resource (in that it extracts the response json) but
POSTs with a body instead of GET.  That method is then
used in ProjectCollection.search.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [x] I have added tests for 100% coverage
- [] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in setup.py

### CODE REVIEW

**Comment by maxhutch** [comment]
I have used this to run the e2e tests (which is what surfaced the bug).  Prior to using search in `find_collection`, the search method was unused in the e2e tests.

File: src/citrine/_session.py [hunk 1]
```diff
@@ -174,6 +174,14 @@ def get_resource(self, path: str, **kwargs) -> dict:
         response = self.checked_get(path, **kwargs)
         return self._extract_response_json(path, response)
 
+    def get_resource_post(self, path: str, json: dict, **kwargs) -> dict:
```
Comment (sparadiso) [inline src/citrine/_session.py:177]: What is the difference between this and `post_resource`? Is this just for a more intuitive method name to use when resource retrieval is a POST?

File: tests/seeding/test_find_or_create.py [hunk 2]
```diff
@@ -63,6 +63,25 @@ def list(self, page: Optional[int] = None, per_page: int = 100):
             else:
                 return self.projects[(page - 1)*per_page:page*per_page]
 
+        def search(self, search_params: Optional[dict] = None, per_page: int = 100):
```
Comment (sparadiso) [inline tests/seeding/test_find_or_create.py:66]: unused?

Comment (maxhutch) [reply tests/seeding/test_find_or_create.py:66]: find_collection calls `search` now, so this is used (and w/o it the tests fail).

File: src/citrine/_session.py (hunk 1 above)
Comment (maxhutch) [reply src/citrine/_session.py]: I just hadn't noticed that the content was the same as `post_resource`.  While there may be value in having different names, I think simplicity wins here.  I removed `get_resource_post` and am calling `post_resource` in search now.

File: src/citrine/resources/project.py [hunk 3]
```diff
@@ -464,6 +464,6 @@ def _fetch_page_search(self, page: Optional[int] = None,
 
         path = self._get_path() + "/search"
 
-        return self._fetch_page(path=path, fetch_func=self.session.checked_post,
+        return self._fetch_page(path=path, fetch_func=self.session.post_resource,
```
Comment (zoesyang) [inline src/citrine/resources/project.py:467]: Oh sweet!

**Pull request merged**